game.rb

```diff
@@ -0,0 +1,100 @@
+# game class
+class Game
+  
+  attr_accessor :column, :main_diagonal, :second_diagonal, :column_matrix
+
+  WIN = 'win'
+  DRAW = 'draw'
+  ON_GAME = 'on_game'
+
+  def initialize
+    @columns = []
+    @main_diagonal = []
+    @second_diagonal = []
+  end
+
+  def status(board)
+    
+    for i in board.table
+      if i.all?{ |inner| inner == 'X' }
+        return WIN
+      elsif i.all? { |inner| inner == 'O' }
+        return WIN
+      end
+    end
+    
+    board_columns = check_columns board
+    for i in board_columns
+      if i.all?{ |inner| inner == 'X' }
+        return WIN
+      elsif i.all? { |inner| inner == 'O' }
+        return WIN
+      end
+    end
+
+    board_diagonal = check_main_diagonal board
+    if board_diagonal.all?{ |inner| inner == ['X'] }
+      return WIN
+    elsif board_diagonal.all? { |inner| inner == ['O'] }
+      return WIN
+    end
+
+    board_diagonal = check_second_diagonal board
+    if board_diagonal.all?{ |inner| inner == ['X'] }
+      return WIN
+    elsif board_diagonal.all? { |inner| inner == ['O'] }
+      return WIN
+    end
+
+    if board.table.flatten.none?{ |inner| inner == nil }
+      return DRAW
+    end
+    
+    ON_GAME
+  end
+
+  private
+
+  def check_columns board
+    column_matrix = []
+    for i in 0...board.table.length
+      for j in 0...board.table.length
+        @columns << board.table[j][i]
+      end
+      column_matrix << @columns
+      @columns = []
+    end
```

Comment:
Here's an idea:
What if you use `map` to generate an array of arrays that arranges the contents of the column in the same fashion as the rows?
Something like:
`[[1, 4, 7], [2,5,8], [3,6,9] ]`

Then you could use the same method that checks that all the elements are the same :D

You could also create a method that returns the diagonal values as arrays 😏